### PR TITLE
[BUG] #14203 fix:Sound serialize 

### PR DESCRIPTION
**Pull request by SubratGuptaCtruh**
https://github.com/BabylonJS/Babylon.js/issues/14203

### CODE REVIEW

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Snapshot stored with reference name:
refs/pull/14204/merge

Test environment: 
https://babylonsnapshots.z22.web.core.windows.net/refs/pull/14204/merge/index.html

To test a playground add it to the URL, for example:

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/14204/merge/index.html#WGZLGJ#4600

Links to test babylon tools with this snapshot:

https://playground.babylonjs.com/?snapshot=refs/pull/14204/merge
https://sandbox.babylonjs.com/?snapshot=refs/pull/14204/merge
https://gui.babylonjs.com/?snapshot=refs/pull/14204/merge
https://nme.babylonjs.com/?snapshot=refs/pull/14204/merge

To test the snapshot in the playground with a playground ID add it after the snapshot query string:

https://playground.babylonjs.com/?snapshot=refs/pull/14204/merge#BCU1XR#0

**Review by sebavan — CHANGES_REQUESTED** [review]
The serialization in case of an array buffer would not work in this case.

**Comment by sebavan** [comment]
Let s add @docEdub to help on this one and ensure of the best fix.

**Comment by sebavan** [comment]
I think it was like this for back compat or to be compatible with files from our exporters  as nothing is serializing url as you  mentions.

File: packages/dev/core/src/Audio/sound.ts [hunk 1]
```diff
@@ -183,7 +183,7 @@ export class Sound {
     private _offset?: number;
     private _tryToPlayTimeout: Nullable<NodeJS.Timeout>;
     private _audioUnlockedObserver?: Nullable<Observer<IAudioEngine>>;
-
+    private _urlOrArrayBuffer:any;
```
Comment (docEdub) [inline packages/dev/core/src/Audio/sound.ts:186]:
```suggestion
    private _url?: Nullable<string>;
```

File: packages/dev/core/src/Audio/sound.ts [hunk 2]
```diff
@@ -1164,7 +1165,7 @@ export class Sound {
     public serialize(): any {
         const serializationObject: any = {
             name: this.name,
-            url: this.name,
+            url: this._urlOrArrayBuffer,
```
Comment (docEdub) [inline packages/dev/core/src/Audio/sound.ts:1168]:
```suggestion
            url: this._url,

```

File: packages/dev/core/src/Audio/sound.ts [hunk 3]
```diff
@@ -201,6 +201,7 @@ export class Sound {
      */
     constructor(name: string, urlOrArrayBuffer: any, scene?: Nullable<Scene>, readyToPlayCallback: Nullable<() => void> = null, options?: ISoundOptions) {
         this.name = name;
+        this._urlOrArrayBuffer=urlOrArrayBuffer;
```
Comment (docEdub) [inline packages/dev/core/src/Audio/sound.ts:204]:
```suggestion
        this._url = urlOrArrayBuffer;
```
... but move it down so it is right after `this._urlType = "String";` so it only gets set if the `urlOrArrayBuffer` argument is a string.

**Review by docEdub — CHANGES_REQUESTED** [review]
Thanks for this PR!

Please make the changes I listed, and we'll get it merged.

**Comment by SubratGuptaCtruh** [comment]
@docEdub  was thinking of also serialising  _urlType and in parser deserialising url into a buffer or url depending on  _urlType , should i go with this approach  , or serialising just the url is good enuf

**Comment by docEdub** [comment]
> @docEdub was thinking of also serialising _urlType and in parser deserialising url into a buffer or url depending on _urlType , should i go with this approach , or serialising just the url is good enuf

Just serialize the url for now. If someone needs it to work with a buffer then we can revisit it.

**Review by docEdub — CHANGES_REQUESTED** [review]
Almost there! Run command `npm run format:fix` to get the checks to pass.

**Review by docEdub — APPROVED** [review]
👍 LGTM

**Review by sebavan — APPROVED** [review]

**Pull request merged**